openssl/src/pkcs7.rs

```diff
@@ -311,11 +398,28 @@ impl Pkcs7Ref {
             Ok(stack)
         }
     }
+
+    /// Retrieve the SignerInfo entries from the PKCS#7 structure.
+    ///
+    /// This corresponds to [`PKCS7_get_signer_info`].
+    ///
+    /// [`PKCS7_get_signer_info`]: https://man.archlinux.org/man/community/libressl/libressl-PKCS7_get_signer_info.3.en
+    pub fn signer_infos(&self) -> Result<&StackRef<Pkcs7SignerInfo>, ErrorStack> {
```

Comment:
From looking at the implementation, this should return an `Option<StackRef<Pkcs7SignerInfo>>` rather than a `Result`: https://github.com/openssl/openssl/blob/c7d848e220ecb432faa05d2198ae689298d71728/crypto/pkcs7/pk7_lib.c#L528.

Reply:
Fixed, thanks!